ostd/src/util/id_set.rs

```diff
@@ -0,0 +1,646 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! A fixed-size set of unique IDs.
+//!
+//! This module introduces two abstract data collection types.
+//! The first one is [`IdSet<I>`],
+//! a set that contains at most `I::cardinality()` items,
+//! each of which represents a unique ID of type `I`.
+//! The second one is [`AtomicIdSet<I>`],
+//! the atomic version of `IdSet<I>`.
+//!
+//! The type parameter `I` implements the [`Id`] trait,
+//! which abstracts any ID type whose instances
+//! can be 1:1 mapped to the integers from 0 to `Id::cardinality()` (exclusive).
+//! The ID type is required to implement `Into<u32>` and `TryFrom<u32>`.
+//!
+//! One use case of `IdSet<CpuId>` in OSTD is [`crate::cpu::set::CpuSet`],
```

Comment:
```suggestion
//! One use case of `IdSet` in OSTD is [`crate::cpu::set::CpuSet`],
```
I don't think you mean `IdSet<CpuId>` here because it is mentioned in the next line (line 18).